x-pack/plugins/triggers_actions_ui/public/types.ts

```diff
@@ -169,14 +174,14 @@ export interface AlertTableItem extends Alert {
 }
 
 export interface AlertTypeParamsExpressionProps<
-  AlertParamsType = unknown,
+  Params extends AlertTypeParams = AlertTypeParams,
   MetaData = Record<string, any>
 > {
-  alertParams: AlertParamsType;
+  alertParams: Params;
   alertInterval: string;
   alertThrottle: string;
   setAlertParams: (property: string, value: any) => void;
-  setAlertProperty: <Key extends keyof Alert>(key: Key, value: Alert[Key] | null) => void;
+  setAlertProperty: <Key extends keyof Alert>(key: Key, value: any) => void;
```

Comment:
Turns out this type was wrong the whole time, and had an underlying `any`.
At least now it's explicit... but I'll still try to fix this before we merge.